server/platform/services/cache/redis.go

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/einterfaces"
+
+	"github.com/redis/rueidis"
+	"github.com/tinylib/msgp/msgp"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const clientSideTTL = 5 * time.Minute
+
+type Redis struct {
+	name          string
+	client        rueidis.Client
+	defaultExpiry time.Duration
+	metrics       einterfaces.MetricsInterface
+}
+
+func NewRedis(opts *CacheOptions, client rueidis.Client) (*Redis, error) {
+	if opts.Name == "" {
+		return nil, errors.New("no name specified for cache")
+	}
+	return &Redis{
+		name:          opts.Name,
+		defaultExpiry: opts.DefaultExpiry,
+		client:        client,
+	}, nil
+}
+
+func (r *Redis) Purge() error {
+	// TODO: move to scan
+	keys, err := r.Keys()
+	if err != nil {
+		return err
+	}
+	return r.client.Do(context.Background(),
```

Comment:
I see we are using `context.Background()`  pretty much everywhere in this package. Given these are network requests underneath I wonder whether we should establish some timeout value or is it already captured by the client itself?

Reply:
Yep, it is captured by the client itself. https://pkg.go.dev/github.com/redis/rueidis#ClientOption - `ConnWriteTimeout`. It is the read/write timeout for each connection.

It was also advised against setting context based timeouts in the earlier client: https://redis.uptrace.dev/guide/go-redis-debugging.html#timeouts